### PR TITLE
(wip) build: getting the frontend build pipeline working with pnpm

### LINKED ISSUE #93: Update frontend build tooling and dependencies
As it stands, Node 16 is out of date and the Dockerfile isn't building. I got it working locally with Node 22 and pnpm. There's going to be a lot of churn + movement in the frontend package, is there any issue with this?

## Build + Environment Changes
- [ ] Update Node to latest LTS
- [ ] Update Dockerfile installation
- [ ] Update GitHub Actions image to latest Node LTS
- [ ] Swap to pnpm instead of yarn

## Frontend Build
After these infra updates, I'd like to upgrade to Vue 3 and Vite instead of Webpack. I'd also like to add TypeScript, if you don't mind.
- [ ] Vue 3 Upgrade
- [ ] Webpack => Vite

### PR DESCRIPTION
**Pull request by JessicaSachs**
Resolves #93 in order to build and work in the frontend package

Let me know if you have questions and can still build the project. I tested locally.

### CODE REVIEW

File: frontend/.npmrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node-options=--openssl-legacy-provider
+node-linker=hoisted
```
Comment (nanmu42) [inline frontend/.npmrc:2]: I add this to address [building issues on Linux](https://github.com/pnpm/pnpm/issues/3585).

Feel free to change it if you have a better solution. :D

**Review by nanmu42 — APPROVED** [review]
Thank you very much, @JessicaSachs .

You may modify and/or merge this PR at your will.

Hope you have a pleasant weekend!

**Comment by JessicaSachs** [comment]
Ah. I didn't realize CI was green. Once this is deployed, let me know so I can feel comfortable working on what's next. 🫶

**Comment by nanmu42** [comment]
Version [v1.8.0](https://github.com/nanmu42/orly/releases/tag/v1.8.0) and [v1.8.1](https://github.com/nanmu42/orly/releases/tag/v1.8.1)(base image name fix) are out and deployed at https://orly.nanmu.me/ .

Docker images: https://hub.docker.com/r/nanmu42/orly

@JessicaSachs , thanks for the wonderful work!

**Pull request merged**